jobs: fix startable job num-runs overcount

samiskin opened the pull request:
Release justification: low risk bug fix

Removes extra call to started() in StartableJob's Start since its
already called in stepThroughStateMachine later on

Release note (bug fix): fix num_runs being incremented twice for certain
jobs upon being started

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/77240)
<!-- Reviewable:end -->

samiskin committed:
jobs: fix startable job num-runs overcount

Release justification: low risk bug fix

Removes extra call to started() in StartableJob's Start since its
already called in stepThroughStateMachine later on

Release note (bug fix): fix num_runs being incremented twice for certain
jobs upon being started

samiskin commented:
bors r+
